openpype/modules/kitsu/kitsu_module.py

```diff
@@ -0,0 +1,477 @@
+"""Kitsu module."""
+
+import click
+import os
+import re
+
+import gazu
+from gazu.asset import all_assets_for_project, all_asset_types, new_asset
+from gazu.shot import (
+    all_episodes_for_project,
+    all_sequences_for_project,
+    all_shots_for_project,
+    new_episode,
+    new_sequence,
+    new_shot,
+)
+from gazu.task import (
+    all_task_types,
+    new_task,
+    new_task_type,
+)
+from pymongo import DeleteOne, UpdateOne
+
+from avalon.api import AvalonMongoDB
+from openpype.api import get_project_settings
+from openpype.modules import JsonFilesSettingsDef, OpenPypeModule, ModulesManager
+from openpype.modules.kitsu.utils.openpype import (
+    create_op_asset,
+    sync_project,
+    update_op_assets,
+)
+from openpype.settings.lib import get_local_settings
+from openpype_interfaces import IPluginPaths, ITrayAction
+from .utils.listeners import start_listeners
+
+
+class AddonSettingsDef(JsonFilesSettingsDef):
+    schema_prefix = "kitsu"
+
+    def get_settings_root_path(self):
+        """Implemented abstract class of JsonFilesSettingsDef.
+
+        Return directory path where json files defying addon settings are
+        located.
+        """
+        return os.path.join(os.path.dirname(os.path.abspath(__file__)), "settings")
+
+
+class KitsuModule(OpenPypeModule, IPluginPaths, ITrayAction):
+    """Kitsu module class."""
+
+    label = "Kitsu"
+    name = "kitsu"
+
+    def initialize(self, settings):
+        """Initialization of module."""
+        module_settings = settings[self.name]
+        local_kitsu_settings = get_local_settings().get("kitsu", {})
+
+        # Enabled by settings
+        self.enabled = module_settings.get("enabled", False)
+
+        # Add API URL schema
+        kitsu_url = module_settings["server"].strip()
+        if kitsu_url:
+            # Ensure web url
+            if not kitsu_url.startswith("http"):
+                kitsu_url = "https://" + kitsu_url
+
+            # Check for "/api" url validity
+            if not kitsu_url.endswith("api"):
+                kitsu_url = f"{kitsu_url}{'' if kitsu_url.endswith('/') else '/'}api"
+
+        self.server_url = kitsu_url
+
+        # Set credentials
+        self.kitsu_login = local_kitsu_settings["login"]
+        self.kitsu_password = local_kitsu_settings["password"]
+
+        # Prepare variables that can be used or set afterwards
+        self._connected_modules = None
+        # UI which must not be created at this time
+        self._dialog = None
+
+    def tray_init(self):
+        """Implementation of abstract method for `ITrayAction`.
+
+        We're definitely  in tray tool so we can pre create dialog.
+        """
+
+        self._create_dialog()
+
+    def get_global_environments(self):
+        """Kitsu's global environments."""
+        return {
+            "KITSU_SERVER": self.server_url,
+            "KITSU_LOGIN": self.kitsu_login,
```

Comment:
How can you know kitsu login if user did not login yet.

Reply:
If I understand you well, the user will have to log-in every time he launches Tray? Doing this was to allow the user to use the `remember` feature for credentials.

Reply:
Fixed